src/components/FilterBar.tsx

```diff
@@ -88,62 +102,26 @@ const FilterBar: React.FC<IFilterBarProps> = (props) => {
           removeAction={handleFilterRemoveClick}
           resetAction={handleFilterResetClick}
         />
-        <FilterBarOptions
-          filters={tagFilters}
-          label="Filter by Tag"
-          allowCollapse={allowCollapse}
-          addAction={handleFilterAddClick}
-        />
-        <FilterBarOptions
-          filters={taskFilters}
-          label="Filter by Task"
-          allowCollapse={allowCollapse}
-          addAction={handleFilterAddClick}
-        />
-        <FilterBarOptions
-          filters={typeFilters}
-          label="Filter by Type"
-          allowCollapse={allowCollapse}
-          addAction={handleFilterAddClick}
-        />
-        <FilterBarOptions
-          filters={licenseFilters}
-          label="Filter by License"
-          allowCollapse={allowCollapse}
-          addAction={handleFilterAddClick}
-        />
-        <FilterBarOptions
-          filters={languageFilters}
-          label="Filter by Language"
-          allowCollapse={allowCollapse}
-          addAction={handleFilterAddClick}
-        />
+        {filterBarOptions}
```

Comment:
NIT: Continued from above. Alternatively to the `for` loop you used, you can get rid of that completely and instead do:

```ts
{filterPropertyNames.map((propName) => {
  <FilterBarOptions
    filters={filterOptions[propName]}
    label={`Filter by ${ProjectFilterProperties[propName]}`}
    allowCollapse={allowCollapse}
    addAction={handleFilterAddClick}
  />;
})}
```

This completely replaces your `{filterBarOptions}` within the JSX return block.

Both approaches are acceptable, but this way is generally the way that is subjectively considered "best" by the wider TypeScript and React communities. In my opinion it does make it easier to read the JSX, but again that's a subjective take on it, and I'll let you make the final decision on how you want to implement this.